### PR TITLE
HTTPS everywhere

### PR DESCRIPTION
**Pull request by barrymcgee**
## Done

- HTTPS everywhere where appropriate
 - Update references to bare domain

## QA

- Check out this feature branch
- Run the site using the command `./run serve`
- View the site locally in your web browser at: [http://0.0.0.0:8001/](http://0.0.0.0:8001/)
- Sanity check code

### CODE REVIEW

File: templates/community/mission.html [hunk 1]
```diff
@@ -60,7 +60,7 @@ <h3>What is open source?</h3>
         <li>The licence must not restrict other software.</li>
         <li>The licence must be technology-neutral.</li>
       </ol>
-      <p><a class="p-link--external" href="http://www.opensource.org/docs/definition.php">Take a look at the Open Source Initiative's definition of open-source software for a more detailed explanation</a>.</p>
+      <p><a class="p-link--external" href="https://www.opensource.org/docs/definition.php">Take a look at the Open Source Initiative's definition of open-source software for a more detailed explanation</a>.</p>
```
Comment (nottrobin) [inline templates/community/mission.html]: The final destination of this link is https://opensource.org/docs/definition.php

File: templates/community/diversity.html [hunk 2]
```diff
@@ -18,7 +18,7 @@ <h1>Diversity</h1>
...
       <p>Although this list cannot be exhaustive, we explicitly honour diversity in age, culture, ethnicity, genotype, gender identity or expression, language, national origin, neurotype, phenotype, political beliefs, profession, race, religion, sexual orientation, socio-economic status, subculture and technical ability.</p>
-      <p>Some of the ideas and wording for this statement were based on diversity statements from the <a class="p-link--external" href="http://www.python.org/community/diversity">Python community</a> and <a class="p-link--external" href="http://www.dreamwidth.org/legal/diversity">Dreamwidth Studios</a> (CC-BY-SA 3.0).</p>
+      <p>Some of the ideas and wording for this statement were based on diversity statements from the <a class="p-link--external" href="https://www.python.org/community/diversity">Python community</a> and <a class="p-link--external" href="https://www.dreamwidth.org/legal/diversity">Dreamwidth Studios</a> (CC-BY-SA 3.0).</p>
```
Comment (nottrobin) [inline templates/community/diversity.html]: https://www.python.org/community/diversity/ should have a slash

File: templates/blog/press-centre.html [hunk 3]
```diff
@@ -46,7 +46,7 @@ <h5 class="p-muted-heading">Facebook</h5>
           </ul>
           <h5 class="p-muted-heading">YouTube</h5>
           <ul class="p-inline-list--middot">
-            <li class="p-inline-list__item"><a href="http://www.youtube.com/celebrateubuntu" title="Follow us on YouTube">Celebrate Ubuntu</a></li>
+            <li class="p-inline-list__item"><a href="https://www.youtube.com/celebrateubuntu" title="Follow us on YouTube">Celebrate Ubuntu</a></li>
```
Comment (nottrobin) [inline templates/blog/press-centre.html]: https://www.youtube.com/user/celebrateubuntu

File: templates/engage/14-04-esm.html [hunk 4]
```diff
@@ -137,7 +137,7 @@ <h3 class="p-heading--four">Sign up for our monthly Cloud&nbsp;Newsletter</h3>
               <input value="30" aria-label="" class="mktoField mktoFieldDescriptor" name="subId" type="hidden">
               <input type="hidden" aria-label="" name="lpurl" value="https://pages.canonical.com/Insights-Subscription_Insights-Subscription-test.html?cr={creative}&amp;kw={keyword}">
               <input value="1212" aria-label="" class="mktoField mktoFieldDescriptor" name="formid" type="hidden">
-              <input type="hidden" aria-label="" name="ret" value="http://ubuntu.com/openstack/newsletter-thank-you">
+              <input type="hidden" aria-label="" name="ret" value="https://www.ubuntu.com/openstack/newsletter-thank-you">
```
Comment (nottrobin) [inline templates/engage/14-04-esm.html]: Remove www

File: templates/internet-of-things/contact-us.html [hunk 5]
```diff
@@ -11,7 +11,7 @@
   intro_text="If you are thinking about building a digital signage product on Ubuntu, please fill in your details below and a member of our team will get in touch.",
   formid="1422",
   lpId="2166",
-  returnURL="http://ubuntu.com/internet-of-things/thank-you?product=digital-signage" %}
+  returnURL="https://www.ubuntu.com/internet-of-things/thank-you?product=digital-signage" %}
```
Comment (nottrobin) [inline templates/internet-of-things/contact-us.html]: Please remove www

File: templates/internet-of-things/contact-us.html [hunk 6]
```diff
@@ -21,7 +21,7 @@
   intro_text="If you would like to know more about Ubuntu for edge gateways please fill in your details below and a member of our team will get in touch.",
   formid="1663",
   lpId="3143",
-  returnURL="http://ubuntu.com/internet-of-things/thank-you?product=gateways" %}
+  returnURL="https://www.ubuntu.com/internet-of-things/thank-you?product=gateways" %}
```
Comment (nottrobin) [inline templates/internet-of-things/contact-us.html]: Please remove www

File: templates/internet-of-things/contact-us.html [hunk 7]
```diff
@@ -31,7 +31,7 @@
   intro_text="If you would like to know more about Ubuntu for Robotics please fill in your details below and a member of our team will get in touch.",
   formid="1650",
   lpId="2838",
-  returnURL="http://ubuntu.com/internet-of-things/thank-you?product=robotics" %}
+  returnURL="https://www.ubuntu.com/internet-of-things/thank-you?product=robotics" %}
```
Comment (nottrobin) [inline templates/internet-of-things/contact-us.html]: Please remove www, and for the following ones

File: templates/robotics.html [hunk 8]
```diff
@@ -367,7 +367,7 @@ <h2>Ready for the next steps?</h2>
 {% with first_item="_iot_developers", second_item="_iot_newsletter_signup", third_item="_iot_further_reading" %}{% include "shared/contextual_footers/_contextual_footer.html" %}{% endwith %}
 
 <!-- Set default Marketo information for contact form below-->
-<div class="u-hide" id="contact-form-container" data-form-location="/shared/forms/interactive/_robotics" data-form-id="1650" data-lp-id="2838" data-return-url="http://ubuntu.com/internet-of-things/thank-you?product=robotics" data-lp-url="https://pages.ubuntu.com/things-contact-us.html">
+<div class="u-hide" id="contact-form-container" data-form-location="/shared/forms/interactive/_robotics" data-form-id="1650" data-lp-id="2838" data-return-url="https://www.ubuntu.com/internet-of-things/thank-you?product=robotics" data-lp-url="https://pages.ubuntu.com/things-contact-us.html">
```
Comment (nottrobin) [inline templates/robotics.html]: Please remove www

File: templates/shared/contextual_footers/_cloud_newsletter.html [hunk 9]
```diff
@@ -23,7 +23,7 @@ <h3 class="p-heading--four">Sign up for our monthly Cloud&nbsp;Newsletter</h3>
         <input value="30" class="mktoField mktoFieldDescriptor" name="subId" type="hidden" />
         <input type="hidden" name="lpurl" value="https://pages.canonical.com/Insights-Subscription_Insights-Subscription-test.html?cr={creative}&amp;kw={keyword}" />
         <input value="1212" class="mktoField mktoFieldDescriptor" name="formid" type="hidden" />
-        <input type="hidden" name="ret" value="http://ubuntu.com/openstack/newsletter-thank-you" />
+        <input type="hidden" name="ret" value="https://www.ubuntu.com/openstack/newsletter-thank-you" />
```
Comment (nottrobin) [inline templates/shared/contextual_footers/_cloud_newsletter.html]: Again, please remove www

**Comment by barrymcgee** [comment]
@nottrobin Is the bare domain a recent thing or is my whole life predicated on a lie?

**Comment by nottrobin** [comment]
Do you mean in internet fashion? Or for ubuntu.com specifically? It's been a thing on ubuntu.com for a good while... Maybe getting on for a year? Not quite sure exactly.

**Comment by barrymcgee** [comment]
@nottrobin On ubuntu.com specifically... my fork is still www.ubuntu.com so that's where Imma direct the blame here.

**Pull request merged**